single-line-diagram-core/src/main/java/com/powsybl/sld/layout/ImplicitCellDetector.java

```diff
@@ -62,13 +62,16 @@ public void detectCells(VoltageLevelGraph graph) {
         exclusionTypes.add(Node.NodeType.FEEDER);
         List<Node.NodeType> stopTypes = new ArrayList<>();
         stopTypes.add(Node.NodeType.BUS);
-        detectCell(graph, stopTypes, exclusionTypes, true, allocatedNodes);
+        List<List<Node>> internCellsNodes = detectCell(graph, stopTypes, exclusionTypes, allocatedNodes);
+        internCellsNodes
+                .forEach(nodes -> graph
+                        .addCell(new InternCell(graph.getNextCellNumber(), nodes, exceptionIfPatternNotHandled)));
```

Comment:
The use of lambda is recommended to have compact readable code, some say that it if it's not on one line you need to go back to the classical syntax :) Maybe a more readable code with lambdas would be `internCellsNodes.stream().map(new InternCell(...)).forEach(graph::addCell)`?

Reply:
beautiful